code/controllers/subsystems/air.dm

```diff
@@ -312,21 +313,20 @@ Total Unsimulated Turfs: [world.maxx*world.maxy*world.maxz - simulated_turf_coun
 		B.c_merge(A)
 		mark_zone_update(A)
 
-/datum/controller/subsystem/air/proc/connect(turf/simulated/A, turf/simulated/B)
+/datum/controller/subsystem/air/proc/connect(turf/A, turf/B)
 	#ifdef ZASDBG
-	ASSERT(istype(A))
+	ASSERT(isturf(A))
 	ASSERT(isturf(B))
 	ASSERT(A.zone)
 	ASSERT(!A.zone.invalid)
-	//ASSERT(B.zone)
 	ASSERT(A != B)
 	#endif
 
-	var/block = air_blocked(A,B)
+	var/block = air_blocked(A,B) || !SHOULD_PARTICIPATE_IN_ZAS(A)
 	if(block & AIR_BLOCKED) return
```

Comment:
Would this trigger in case of `!SHOULD_PARTICIPATE_IN_ZAS(A)`? Im not sure what TRUE looks on in bit math

Reply:
`!SHOULD_PARTICIPATE_IN_ZAS` -> `AIR_BLOCKED`. It might need to be `var/block = air_blocked(A,B) || (SHOULD_PARTICIPATE_IN_ZAS(A) && BLOCKED)` or something like that.